analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -72,69 +63,148 @@ if (!("GGally" %in% installed.packages())) {
+```
 
-# Make caller specific plots folder
-if (!dir.exists(consensus_results_dir)) {
-  dir.create(consensus_results_dir)
-}
-# Make caller specific plots folder
-if (!dir.exists(consensus_plots_dir)) {
-  dir.create(consensus_plots_dir)
-}
+Note what columns we will join by.
+
+```{r}
+join_cols = c("Chromosome",
+              "Start_Position",
+              "Reference_Allele",
+              "Allele",
+              "Tumor_Sample_Barcode")
+```
+
+Set up tables from the database but only call the columns we need.
+
+```{r}
+strelka <- dplyr::tbl(con, "strelka") %>% 
+  dplyr::select(join_cols, "VAF")
+
+lancet <- dplyr::tbl(con, "lancet") %>% 
+  dplyr::select(join_cols, "VAF")
+
+mutect <- dplyr::tbl(con, "mutect") %>% 
+  dplyr::select(join_cols, "VAF") 
+
+vardict <- dplyr::tbl(con, "vardict") %>% 
+  dplyr::select(join_cols, "VAF") 
 ```
 
-## Import data 
+```{r}
+# Designate consensus table
+consensus <- dplyr::tbl(con, "consensus")
+```
 
-Load in the merged data.frames from `03-merge_callers.R` output. 
+Because `DBI` does not support full_join, we had to use a series of `left_join` and
+`union_all` calls in order to get a full join of all the callers. 
 
 ```{r}
-# Master VAF data.frame
-vaf_df <- readr::read_rds(file.path(consensus_results_dir, 
-                                    "all_callers_vaf.rds"))
+# This script will do the full join of the data for us
+source(file.path("util", "full_join_callers.R"))
+
+# Check that this looks right
+head(all_caller)
+```
 
-# Master TMB data.frame
-tmb_df <- readr::read_rds(file.path(consensus_results_dir, 
-                                    "all_callers_tmb.rds"))
+```{r}
+# Set up the data as a data.frame
+vaf_df <- all_caller %>% 
+  dplyr::select("VAF_lancet", "VAF_strelka", "VAF_vardict", "VAF_mutect") %>%
+  as.data.frame() %>%
+  dplyr::mutate(VAF_consensus = VAF_strelka)     
+ 
+# Which should be part of the consensus
+consensus_index <- which(apply(!is.na(vaf_df[, -3]), 1, sum) < 3)
+ 
+# Make non consensus VAF into an NA
+vaf_df$VAF_consensus[consensus_index] <- NA
+
+# Make this long form for plotting and 
+long_vaf_df <- vaf_df %>% 
+  dplyr::mutate(index = factor(1:nrow(.))) %>% 
+  tidyr::gather(key = "caller", value = "vaf", -index) %>% 
+  dplyr::mutate(caller = gsub("VAF_", "", caller)) %>% 
+  dplyr::filter(!is.na(vaf))
+```
 
-# Full mutation list
-mutation_id_list <- readr::read_rds(file.path(consensus_results_dir, 
-                                              "mutation_id_list.rds"))
+Get median vaf for each mutation with each caller combination. 
 
-# Callers per mutation data.frame
-callers_per_mutation <- readr::read_rds(file.path(consensus_results_dir, 
-                                                  "callers_per_mutation.rds"))
+```{r}
+# Take out the consensus VAF for this combination
+long_caller_df <- long_vaf_df %>% 
+  dplyr::filter(caller != "consensus")
+
+# Determine the combination of callers for each mutation
+callers_per_mutation <- tapply(
```

Comment:
Just out of curiousity, is `tapply` faster than `group_by %>% summarize` here? I feel like we had this discussion earlier, but since you are doing the tapply twice, and you could do both callers and median in the same `summarize` call and not have to worry about conversions, etc, this would DRY it up a bit.

Reply:
It is much faster which is why I have it done this way. This is something we talked about when I originally made these analyses.

Reply:
But I agree, this is ugly.

Reply:
You might be able to speed it up a bit by making one call to `tapply` with a custom function that calculates both `med_vaf` and assembles the `caller` names. Not sure it is worth it, though.

Reply:
Oh. True. I will see how clunk/not clunky that is. Right now I feel like this whole notebook is pretty clunky, so I'm trying to cut down on the clunk.

Reply:
🤣